assisted-inject-processor/src/main/java/com/squareup/inject/assisted/processor/AssistedInjectProcessor.kt

```diff
@@ -178,32 +177,34 @@ class AssistedInjectProcessor : AbstractProcessor() {
    * - At least one provided parameter and no duplicates
    * - Factory method parameters match assisted parameters in any order
    */
-  private fun AssistedInjectElements.toInjectionOrNull(): AssistedInjection? {
+  private fun AssistedInjectElements.toAssistedInjectionOrNull(): AssistedInjection? {
     var valid = true
 
     val requests = targetConstructor.parameters.map { it.asDependencyRequest() }
     val (assistedRequests, providedRequests) = requests.partition { it.isAssisted }
-    if (providedRequests.isEmpty()) {
-      error("Assisted injection requires at least one non-@Assisted parameter.", targetConstructor)
-      valid = false
-    }
     if (assistedRequests.isEmpty()) {
       error("Assisted injection requires at least one @Assisted parameter.", targetConstructor)
       valid = false
+    } else {
```

Comment:
Same code just now grouped the empty/non-empty cases into if/else. Same for provided below.